internal/provider/jwt_issuers_test.go

```diff
@@ -0,0 +1,364 @@
+		p.service = NewService(cl)
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("no ID is set")
+		}
+
+		traceAPICall("GetJWTIssuer")
+		jwtIssuerResp, _, err := p.service.GetJWTIssuer(context.TODO(), rs.Primary.ID)
+		if err == nil {
+			respIdentityMap := *jwtIssuerResp.IdentityMap
+			if jwtIssuerResp.IssuerUrl == issuerURL &&
+				jwtIssuerResp.Audience == testAudience &&
+				*jwtIssuerResp.Jwks == testJWKS &&
+				*jwtIssuerResp.Claim == testClaim &&
+				len(respIdentityMap) == 2 &&
+				*respIdentityMap[0].TokenIdentity == testIdentityMap[0].TokenIdentity.ValueString() &&
+				*respIdentityMap[0].CcIdentity == testIdentityMap[0].CcIdentity.ValueString() &&
+				*respIdentityMap[0].IsRegex == testIdentityMap[0].IsRegex.ValueBool() &&
+				*respIdentityMap[1].TokenIdentity == testIdentityMap[1].TokenIdentity.ValueString() &&
+				*respIdentityMap[1].CcIdentity == testIdentityMap[1].CcIdentity.ValueString() &&
+				*respIdentityMap[1].IsRegex == testIdentityMap[1].IsRegex.ValueBool() {
+				return nil
+			}
+		}
+
+		return fmt.Errorf("JWT Issuer configuration does not have correct values")
+	}
+}
+
+func testJWTIssuerUpdated(resourceName, issuerURL string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		p := testAccProvider.(*provider)
+		p.service = NewService(cl)
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("no ID is set")
+		}
+
+		traceAPICall("GetJWTIssuer")
+		jwtIssuerResp, _, err := p.service.GetJWTIssuer(context.TODO(), rs.Primary.ID)
+		if err == nil {
+			respIdentityMap := *jwtIssuerResp.IdentityMap
+			if jwtIssuerResp.IssuerUrl == issuerURL &&
+				jwtIssuerResp.Audience == testAudienceUpdated &&
+				*jwtIssuerResp.Jwks == testJWKSUpdated &&
+				*jwtIssuerResp.Claim == testClaimUpdated &&
+				len(respIdentityMap) == 1 &&
+				*respIdentityMap[0].TokenIdentity == testIdentityMapUpdated[0].TokenIdentity.ValueString() &&
+				*respIdentityMap[0].CcIdentity == testIdentityMapUpdated[0].CcIdentity.ValueString() &&
+				*respIdentityMap[0].IsRegex == testIdentityMapUpdated[0].IsRegex.ValueBool() {
+				return nil
+			}
+		}
+
+		return fmt.Errorf("JWT Issuer configuration does not have correct values")
+	}
+}
+
+func testJWTIssuerDestroy(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		p := testAccProvider.(*provider)
+		p.service = NewService(cl)
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("no ID is set")
+		}
+
+		traceAPICall("GetJWTIssuer")
+		if _, _, err := p.service.GetJWTIssuer(context.TODO(), rs.Primary.ID); err == nil {
+			return fmt.Errorf("JWT Issuer not deleted")
```

Comment:
I'm not quite following what's happening here.  I'm assuming the api should be returning a 404 if the issuer is deleted?  What does that look like from the perspective of the SDK?

Reply:
As part of resource destroy, we expect the TF Provider to issue a `DELETE` call to Console.
This bit validates that calling the `GetIssuer` does not return any issuer (expects it to be destroyed).
Updated to use HTTP status codes for clarity.